subiquity/models/subiquity.py

```diff
@@ -61,13 +74,15 @@ def __init__(self, common):
     def _cloud_init_config(self):
         user = self.identity.user
         users_and_groups_path = (
-            os.path.join(os.environ.get("SNAP", "/does-not-exist"),
+            os.path.join(os.environ.get("SNAP", "."),
                          "users-and-groups"))
         if os.path.exists(users_and_groups_path):
             groups = open(users_and_groups_path).read().split()
         else:
             groups = ['admin']
         groups.append('sudo')
+        all_groups = get_all_groups(self.opts.dry_run)
+        groups = [group for group in groups if group in all_groups]
```

Comment:
You can put the call to get_all_groups in the list comprehension unless there is a need to keep the all_groups around.

Reply:
Sure.